operator/src/main/java/oracle/kubernetes/operator/DomainProcessorImpl.java

```diff
@@ -706,9 +769,24 @@ private void internalMakeRightDomainPresence() {
     }
 
     private StepAndPacket createDomainPlanSteps(Packet packet) {
+      if (containsAbortedEventData()) {
+        return new StepAndPacket(Step.chain(createEventStep(eventData), new TailStep()), packet);
+      }
+
       return new StepAndPacket(
-          createPopulatePacketServerMapsStep(createSteps()),
-          packet);
+          getEventStep(Step.chain(createPopulatePacketServerMapsStep(),  createSteps())), packet);
+    }
+
+    private Step getEventStep(Step next) {
+      if (eventData != null) {
+        Step step = Step.chain(createEventStep(eventData), next);
+        return step;
+      }
+      return next;
+    }
+
```

Comment:
Here, too, Optional is designed for cases like this. You could write:
```
    return Optional.ofNullable(eventData).map(EventData::isDomainProcessingAborted).orElse(false);
```
if you added the appropriate method to EventData.